internal/agent/TUIinstallProcessPage.go

```diff
@@ -38,84 +46,148 @@ func newInstallProcessPage() *installProcessPage {
 			InstallAfterInstallStep,
 			InstallCompleteStep,
 		},
-		done:   make(chan bool),
-		output: make(chan string),
+		done:          make(chan bool),
+		output:        make(chan string, 10), // Buffered channel to avoid blocking
+		logsDone:      make(chan bool),
+		installerDone: make(chan bool),
 	}
 }
 
 func (p *installProcessPage) Init() tea.Cmd {
-	oldLog := mainModel.log
-	cc := NewInteractiveInstallConfig(&mainModel)
-	// Create a new logger to track the install process output
-	// TODO: Maybe do a dual logger or something? So we can still see the output in the old logger decently
-	logBuffer := bytes.Buffer{}
-	bufferLog := types.NewBufferLogger(&logBuffer)
-	cc.Logger = bufferLog
-
-	// Start the installer in a goroutine
-	go func() {
-		defer close(p.done)
-		err := RunInstall(cc)
-		if err != nil {
+	p.once.Do(func() {
+		internalLogger.Logger.Info().Msg("[Init] Starting installer and log goroutines")
+		oldLog := mainModel.log
```

Comment:
why is this called "old"? Where does the old logger print its messages? I see we print more things to internalLogger. We should use names that reflect the purpose of each logger. E.g. debugLogger and <whatever-the-old-logger-does>Logger 

As far as I can tell, there is no way to see the logs while installation is happening. What is the suggested user flow for monitoring the logs?

Reply:
oldLog is the main config log which logs to file and journalctl. We cant use that one to track the install, so we store it and crete a new one that logs to a buffer so we can go over the buffer lines and know at which point of the install we are and if there is errors. The old logger is used to forward the lines we process from the new buffered logger so the logs end up in the journal as well, otherwise we would lose those logs.

There is a comment around there about using either a dual logger or something similar but thats not implemented as we need both a log that we can go over and to store the logs in persistent storage.

Sucks but this was like this before, it was a quick workaround for the current need.

Reply:
We can send them to `info` log level. At the end of the day, if something doesn't work, it's still the user that will collect logs and send them over to us. So, even if they don't read and understand the logs, they still might care to collect those. I'd say, it's better to stick to one logger for everything.